src/claimtrie.cpp

```diff
@@ -1112,20 +1263,23 @@ bool CClaimTrie::WriteToDisk()
 
 bool CClaimTrie::InsertFromDisk(const std::string& name, CClaimTrieNode* node)
 {
-    if (name.size() == 0)
-    {
+    if (name.size() == 0) {
         root = *node;
         return true;
     }
+    const std::string& newName = (fNormalize ? normalizeClaimName(name) : name);
+    if (fNormalize && newName != name) {
+        delete node;
+        node = new CClaimTrieNode();
+    }
     CClaimTrieNode* current = &root;
-    for (std::string::const_iterator itname = name.begin(); itname + 1 != name.end(); ++itname)
-    {
+    for (std::string::const_iterator itname = newName.begin(); itname + 1 != newName.end(); ++itname) {
         nodeMapType::iterator itchild = current->children.find(*itname);
         if (itchild == current->children.end())
             return false;
```

Comment:
I know this is a preexisting line, but it seems to make a harsh assumption about the order that data in the trie is read from disk (it must be "pre-order").